File: server/src/main/java/server/service/CompanyService.java
package server.service;

import server.entity.Company;

import java.util.List;
import java.util.Optional;

public interface CompanyService {

    Company saveCompany(Company company);

    List<Company> getAllCompanies();

    Optional<Company> getCompanyById(String id);

    Company updateCompany(String id, Company company);

    void deleteCompany(String id);
}

File: client/src/services/companyService.js
import api from "../api/api.js";

export const saveCompany = async (company) => {
    const response = await api.post("/company", company);
    return response.data;
};

export const getCompanies = async () => {
    const response = await api.get("/company");
    return response.data;
};

export const getCompanyById = async (id) => {
    const response = await api.get(`/company/${id}`);
    return response.data;
};

export const updateCompany = async (id, company) => {
    const response = await api.put(`/company/${id}`, company);
    return response.data;
};

export const deleteCompany = async (id) => {
    await api.delete(`/company/${id}`);
};